src/renderer/components/dock/terminal/terminal.ts

```diff
@@ -63,15 +59,25 @@ export class Terminal {
     const { elem } = this;
 
     if (elem) {
-      Terminal.spawningPool.appendChild(elem);
+      this.dependencies.spawningPool.appendChild(elem);
```

Comment:
Big nit: A full `htmlElement` as dependency is a breach of Interface Segregation Principle. If really needed (ie. without portals), the dependency could be `addElementToSpawningPool` or perhaps `mountTerminal`, being more informative.

Reply:
The thing is that `xterm` instances require a full `HTMLElement` to mount onto